launch.go

```diff
@@ -72,6 +73,32 @@ func init() {
 	client = retryablehttp.NewClient()
 }
 
+/* has HTTP or HTTPS protocol
+targetUrl => URL
+*/
+func hasHTTPProtocol(targetUrl *url.URL) bool {
+	return strings.Contains(targetUrl.Scheme, "http") || strings.Contains(targetUrl.Scheme, "https")
+}
+
+/* make pushgateway url
+baseUrl => base url for pushgateway
+buildID => sd build id
+*/
+func makePushgatewayUrl(baseUrl string, buildID int) (*url.URL, error) {
```

Comment:
From a lint perspective, `Url` should be `URL`.

Reply:
I fixed.